shutdown/stop/stop.go

```diff
@@ -48,6 +48,10 @@ func stop(stoppableItems []Stoppable, listener Listener) {
 		serviceName := stoppable.String()
 		listener.ServiceWillBeStopped(serviceName)
 		err := stoppable.Stop()
-		listener.ServiceWasStopped(serviceName, err)
+		if err != nil {
```

Comment:
why is this check needed.
the `ServiceWasStopped` method can handle the error nil case